mongo/src/main/java/org/axonframework/extensions/mongo/eventsourcing/tokenstore/MongoTokenStore.java

```diff
@@ -110,7 +112,27 @@ public static Builder builder() {
 
     @Override
     public void storeToken(TrackingToken token, String processorName, int segment) throws UnableToClaimTokenException {
-        updateOrInsertTokenEntry(token, processorName, segment);
+        updateToken(token, processorName, segment);
+    }
+
+    private void updateToken(TrackingToken token, String processorName, int segment) {
+        AbstractTokenEntry<?> tokenEntry =
+                new GenericTokenEntry<>(token, serializer, contentType, processorName, segment);
+        tokenEntry.claim(nodeId, claimTimeout);
+
+        Bson update = combine(set("owner", nodeId),
+                              set("timestamp", tokenEntry.timestamp().toEpochMilli()),
+                              set("token", tokenEntry.getSerializedToken().getData()),
+                              set("tokenType", tokenEntry.getSerializedToken().getType().getName()));
+
+        UpdateResult updateResult = mongoTemplate.trackingTokensCollection()
+                                                 .updateOne(claimableTokenEntryFilter(processorName, segment), update);
+
+        if (updateResult.getModifiedCount() == 0) {
+            throw new UnableToClaimTokenException(format(
+                    "Unable to claim token '%s[%s]'. It has not been initialized yet", processorName, segment
```

Comment:
Technically, this could also be rejected when the token is initialized, but has been claimed by another node. The problem here is mainly that the current processor doesn't hold a valid claim on the token.

Reply:
Fair point, I'll rephrase the message to `Unable to claim token '%s[%s]'. It is either already claimed or it does not exist`